templates/patient/demographics/relation_form.html.twig

```diff
@@ -217,213 +218,349 @@
-            let badgesHtml = '';
-            if (emergencyCheckbox && emergencyCheckbox.checked) {
-                badgesHtml += '<span class="badge badge-danger mr-1">' + window.xl('Emergency') + '</span>';
+            function toggleRelationEdit(event) {
+                const card = event.target.closest('.relation-card');
+                const formDiv = card.querySelector('.form_relations');
+                const icon = card.querySelector('.btn-edit-relation');
+
+                if (formDiv.classList.contains('d-none')) {
+                    formDiv.classList.remove('d-none');
+                    icon.classList.remove('fa-caret-right');
+                    icon.classList.add('fa-caret-down');
+                } else {
+                    formDiv.classList.add('d-none');
+                    icon.classList.remove('fa-caret-down');
+                    icon.classList.add('fa-caret-right');
+                }
             }
-            if (primaryCheckbox && primaryCheckbox.checked) {
-                badgesHtml += '<span class="badge badge-primary mr-1">' + window.xl('Primary') + '</span>';
+
+            function changeRelationStatus(event) {
+                event.preventDefault();
+                event.stopPropagation();
+
+                const relationsGroup = event.target.closest('.relations_group');
+                const currentActive = relationsGroup.getAttribute('data-active') === '1';
+                const activeInput = relationsGroup.querySelector('input.form_relations_active');
+                const actionInput = relationsGroup.querySelector('input.form_relations_data_action');
+
+                if (currentActive) {
+                    const confirmMsg = window.xl("Are you sure you wish to inactivate this relationship?");
+                    if (!confirm(confirmMsg)) {
+                        return;
+                    }
+                    relationsGroup.setAttribute('data-active', '0');
+                    activeInput.value = '0';
+                    relationsGroup.classList.add('relation-inactive');
+
+                    if (actionInput.value === 'ADD') {
+                        actionInput.value = 'ADD';
+                    } else {
+                        actionInput.value = 'INACTIVATE';
+                    }
+                } else {
+                    relationsGroup.setAttribute('data-active', '1');
+                    activeInput.value = '1';
+                    relationsGroup.classList.remove('relation-inactive');
+                    actionInput.value = 'UPDATE';
+                }
+
+                relationUpdated();
             }
-            displayBadges.innerHTML = badgesHtml;
-        }
-    }
 
-    (function() {
-        'use strict';
+            function relationUpdated(event) {
+                console.log('Relation updated event');
+                const formTelecoms = event.target.closest('.form_telecoms');
+                if (formTelecoms !== null) return;
```

Comment:
@psoas You need to pass the click event to the function relationUpdated(event) from the calling locations(2 calls, here and search)
Even still, fails to update state of widget after I fixed.

Also noted for the show Inactive check box in Related view for inactive persons no click event so unsure how it's supposed to work only that it doesn't.